File: app/src/main/java/com/sasfmlzr/findfilm/request/CurrentMovieRequest.java
package com.sasfmlzr.findfilm.request;

import android.graphics.Bitmap;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

@SuppressWarnings({"unused"})
public class CurrentMovieRequest {
    @SerializedName("adult")
    @Expose
    private boolean adult;
    @SerializedName("backdrop_path")
    @Expose
    private String backdropPath;
    @SerializedName("belongs_to_collection")
    @Expose
    private Object belongsToCollection;
    @SerializedName("budget")
    @Expose
    private int budget;
    @SerializedName("genres")
    @Expose
    private List<Genre> genres = null;
    @SerializedName("homepage")
    @Expose
    private String homepage;
    @SerializedName("id")
    @Expose
    private int id;
    @SerializedName("imdb_id")
    @Expose
    private String imdbId;
    @SerializedName("original_language")
    @Expose
    private String originalLanguage;
    @SerializedName("original_title")
    @Expose
    private String originalTitle;
    @SerializedName("overview")
    @Expose
    private String overview;
    @SerializedName("popularity")
    @Expose
    private double popularity;
    @SerializedName("poster_path")
    @Expose
    private Object posterPath;
    @SerializedName("production_companies")
    @Expose
    private List<ProductionCompany> productionCompanies = null;
    @SerializedName("production_countries")
    @Expose
    private List<ProductionCountry> productionCountries = null;
    @SerializedName("release_date")
    @Expose
    private String releaseDate;
    @SerializedName("revenue")
    @Expose
    private int revenue;
    @SerializedName("runtime")
    @Expose
    private int runtime;
    @SerializedName("spoken_languages")
    @Expose
    private List<SpokenLanguage> spokenLanguages = null;
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("tagline")
    @Expose
    private String tagline;
    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("video")
    @Expose
    private boolean video;
    @SerializedName("vote_average")
    @Expose
    private double voteAverage;
    @SerializedName("vote_count")
    @Expose
    private int voteCount;
    private Bitmap backdropBitmap;

    public CurrentMovieRequest() {
    }

    public CurrentMovieRequest(boolean adult, String backdropPath, Object belongsToCollection, int budget, List<Genre> genres, String homepage, int id, String imdbId, String originalLanguage, String originalTitle, String overview, double popularity, Object posterPath, List<ProductionCompany> productionCompanies, List<ProductionCountry> productionCountries, String releaseDate, int revenue, int runtime, List<SpokenLanguage> spokenLanguages, String status, String tagline, String title, boolean video, double voteAverage, int voteCount) {
        super();
        this.adult = adult;
        this.backdropPath = backdropPath;
        this.belongsToCollection = belongsToCollection;
        this.budget = budget;
        this.genres = genres;
        this.homepage = homepage;
        this.id = id;
        this.imdbId = imdbId;
        this.originalLanguage = originalLanguage;
        this.originalTitle = originalTitle;
        this.overview = overview;
        this.popularity = popularity;
        this.posterPath = posterPath;
        this.productionCompanies = productionCompanies;
        this.productionCountries = productionCountries;
        this.releaseDate = releaseDate;
        this.revenue = revenue;
        this.runtime = runtime;
        this.spokenLanguages = spokenLanguages;
        this.status = status;
        this.tagline = tagline;
        this.title = title;
        this.video = video;
        this.voteAverage = voteAverage;
        this.voteCount = voteCount;
    }

    public boolean isAdult() {
        return adult;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public Object getBelongsToCollection() {
        return belongsToCollection;
    }

    public int getBudget() {
        return budget;
    }

    public List<Genre> getGenres() {
        return genres;
    }

    public String getHomepage() {
        return homepage;
    }

    public int getId() {
        return id;
    }

    public String getImdbId() {
        return imdbId;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public String getOriginalTitle() {
        return originalTitle;
    }

    public String getOverview() {
        return overview;
    }

    public double getPopularity() {
        return popularity;
    }

    public Object getPosterPath() {
        return posterPath;
    }

    public List<ProductionCompany> getProductionCompanies() {
        return productionCompanies;
    }

    public List<ProductionCountry> getProductionCountries() {
        return productionCountries;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public int getRevenue() {
        return revenue;
    }

    public int getRuntime() {
        return runtime;
    }

    public List<SpokenLanguage> getSpokenLanguages() {
        return spokenLanguages;
    }

    public String getStatus() {
        return status;
    }

    public String getTagline() {
        return tagline;
    }

    public String getTitle() {
        return title;
    }

    public boolean isVideo() {
        return video;
    }

    public double getVoteAverage() {
        return voteAverage;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public Bitmap getBackdropBitmap() {
        return backdropBitmap;
    }

    public void setBackdropBitmap(Bitmap backdropBitmap) {
        this.backdropBitmap = backdropBitmap;
    }

    public static class Genre {

        @SerializedName("id")
        @Expose
        private int id;
        @SerializedName("name")
        @Expose
        private String name;

        public Genre() {
        }

        public Genre(int id, String name) {
            super();
            this.id = id;
            this.name = name;
        }

        public int getId() {
            return id;
        }

        public String getName() {
            return name;
        }
    }

    public static class ProductionCompany {

        @SerializedName("id")
        @Expose
        private int id;
        @SerializedName("logo_path")
        @Expose
        private String logoPath;
        @SerializedName("name")
        @Expose
        private String name;
        @SerializedName("origin_country")
        @Expose
        private String originCountry;

        public ProductionCompany() {
        }

        public ProductionCompany(int id, String logoPath, String name, String originCountry) {
            super();
            this.id = id;
            this.logoPath = logoPath;
            this.name = name;
            this.originCountry = originCountry;
        }

        public int getId() {
            return id;
        }

        public String getLogoPath() {
            return logoPath;
        }

        public String getName() {
            return name;
        }

        public String getOriginCountry() {
            return originCountry;
        }
    }

    public static class ProductionCountry {

        @SerializedName("iso_3166_1")
        @Expose
        private String iso31661;
        @SerializedName("name")
        @Expose
        private String name;

        public ProductionCountry() {
        }

        public ProductionCountry(String iso31661, String name) {
            super();
            this.iso31661 = iso31661;
            this.name = name;
        }

        public String getIso31661() {
            return iso31661;
        }

        public String getName() {
            return name;
        }
    }

    public static class SpokenLanguage {

        @SerializedName("iso_639_1")
        @Expose
        private String iso6391;
        @SerializedName("name")
        @Expose
        private String name;

        public SpokenLanguage() {
        }

        public SpokenLanguage(String iso6391, String name) {
            super();
            this.iso6391 = iso6391;
            this.name = name;
        }

        public String getIso6391() {
            return iso6391;
        }

        public String getName() {
            return name;
        }
    }
}

File: app/src/main/java/com/sasfmlzr/findfilm/request/CurrentMovieRequest.kt
package com.sasfmlzr.findfilm.request

import android.graphics.Bitmap
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class CurrentMovieRequest {
    @SerializedName("adult")
    @Expose
    var isAdult = false
        private set

    @SerializedName("backdrop_path")
    @Expose
    var backdropPath: String? = null
        private set

    @SerializedName("belongs_to_collection")
    @Expose
    var belongsToCollection: Any? = null
        private set

    @SerializedName("budget")
    @Expose
    var budget = 0
        private set

    @SerializedName("genres")
    @Expose
    var genres: List<Genre>? = null
        private set

    @SerializedName("homepage")
    @Expose
    var homepage: String? = null
        private set

    @SerializedName("id")
    @Expose
    var id = 0
        private set

    @SerializedName("imdb_id")
    @Expose
    var imdbId: String? = null
        private set

    @SerializedName("original_language")
    @Expose
    var originalLanguage: String? = null
        private set

    @SerializedName("original_title")
    @Expose
    var originalTitle: String? = null
        private set

    @SerializedName("overview")
    @Expose
    var overview: String? = null
        private set

    @SerializedName("popularity")
    @Expose
    var popularity = 0.0
        private set

    @SerializedName("poster_path")
    @Expose
    var posterPath: Any? = null
        private set

    @SerializedName("production_companies")
    @Expose
    var productionCompanies: List<ProductionCompany>? = null
        private set

    @SerializedName("production_countries")
    @Expose
    var productionCountries: List<ProductionCountry>? = null
        private set

    @SerializedName("release_date")
    @Expose
    var releaseDate: String? = null
        private set

    @SerializedName("revenue")
    @Expose
    var revenue = 0.0
        private set

    @SerializedName("runtime")
    @Expose
    var runtime = 0
        private set

    @SerializedName("spoken_languages")
    @Expose
    var spokenLanguages: List<SpokenLanguage>? = null
        private set

    @SerializedName("status")
    @Expose
    var status: String? = null
        private set

    @SerializedName("tagline")
    @Expose
    var tagline: String? = null
        private set

    @SerializedName("title")
    @Expose
    var title: String? = null
        private set

    @SerializedName("video")
    @Expose
    var isVideo = false
        private set

    @SerializedName("vote_average")
    @Expose
    var voteAverage = 0.0
        private set

    @SerializedName("vote_count")
    @Expose
    var voteCount = 0
        private set
    var backdropBitmap: Bitmap? = null

    constructor() {}
    constructor(
        adult: Boolean,
        backdropPath: String?,
        belongsToCollection: Any?,
        budget: Int,
        genres: List<Genre>?,
        homepage: String?,
        id: Int,
        imdbId: String?,
        originalLanguage: String?,
        originalTitle: String?,
        overview: String?,
        popularity: Double,
        posterPath: Any?,
        productionCompanies: List<ProductionCompany>?,
        productionCountries: List<ProductionCountry>?,
        releaseDate: String?,
        revenue: Int,
        runtime: Int,
        spokenLanguages: List<SpokenLanguage>?,
        status: String?,
        tagline: String?,
        title: String?,
        video: Boolean,
        voteAverage: Double,
        voteCount: Int
    ) : super() {
        isAdult = adult
        this.backdropPath = backdropPath
        this.belongsToCollection = belongsToCollection
        this.budget = budget
        this.genres = genres
        this.homepage = homepage
        this.id = id
        this.imdbId = imdbId
        this.originalLanguage = originalLanguage
        this.originalTitle = originalTitle
        this.overview = overview
        this.popularity = popularity
        this.posterPath = posterPath
        this.productionCompanies = productionCompanies
        this.productionCountries = productionCountries
        this.releaseDate = releaseDate
        this.revenue = revenue.toDouble()
        this.runtime = runtime
        this.spokenLanguages = spokenLanguages
        this.status = status
        this.tagline = tagline
        this.title = title
        isVideo = video
        this.voteAverage = voteAverage
        this.voteCount = voteCount
    }

    class Genre {
        @SerializedName("id")
        @Expose
        var id = 0
            private set

        @SerializedName("name")
        @Expose
        var name: String? = null
            private set

        constructor() {}
        constructor(id: Int, name: String?) : super() {
            this.id = id
            this.name = name
        }
    }

    class ProductionCompany {
        @SerializedName("id")
        @Expose
        var id = 0
            private set

        @SerializedName("logo_path")
        @Expose
        var logoPath: String? = null
            private set

        @SerializedName("name")
        @Expose
        var name: String? = null
            private set

        @SerializedName("origin_country")
        @Expose
        var originCountry: String? = null
            private set

        constructor() {}
        constructor(id: Int, logoPath: String?, name: String?, originCountry: String?) : super() {
            this.id = id
            this.logoPath = logoPath
            this.name = name
            this.originCountry = originCountry
        }
    }

    class ProductionCountry {
        @SerializedName("iso_3166_1")
        @Expose
        var iso31661: String? = null
            private set

        @SerializedName("name")
        @Expose
        var name: String? = null
            private set

        constructor() {}
        constructor(iso31661: String?, name: String?) : super() {
            this.iso31661 = iso31661
            this.name = name
        }
    }

    class SpokenLanguage {
        @SerializedName("iso_639_1")
        @Expose
        var iso6391: String? = null
            private set

        @SerializedName("name")
        @Expose
        var name: String? = null
            private set

        constructor() {}
        constructor(iso6391: String?, name: String?) : super() {
            this.iso6391 = iso6391
            this.name = name
        }
    }
}